oracle-plugin/docs/Oracle-batchsink.md

```diff
@@ -38,6 +38,38 @@ will be passed to the JDBC driver as connection arguments for JDBC drivers that
 
 **Default Batch Value:** The default batch value that triggers an execution request.
 
+
+Data Types Mapping
+----------
+
+    | Oracle Data Type               | CDAP Schema Data Type | Comment                |
+    | ------------------------------ | --------------------- | ---------------------- |
+    | VARCHAR2                       | string                |                        |
+    | NVARCHAR2                      | string                |                        |
+    | VARCHAR                        | string                |                        |
+    | NUMBER                         | decimal               |                        |
+    | FLOAT                          | double                |                        |
+    | LONG                           | string                |                        |
+    | DATE                           | timestamp             |                        |
+    | BINARY_FLOAT                   | float                 |                        |
+    | BINARY_DOUBLE                  | double                |                        |
+    | TIMESTAMP                      | timestamp             |                        |
+    | TIMESTAMP WITH TIME ZONE       | timestamp             | Converted to UTC time  |
+    | TIMESTAMP WITH LOCAL TIME ZONE | timestamp             |                        |
+    | INTERVAL YEAR TO MONTH         | string                |                        |
+    | INTERVAL DAY TO SECOND         | string                |                        |
+    | RAW                            | bytes                 |                        |
+    | LONG RAW                       | bytes                 |                        |
+    | ROWID                          | string                |                        |
+    | UROWID                         | string                |                        |
+    | CHAR                           | string                |                        |
+    | NCHAR                          | string                |                        |
+    | CLOB                           | string                |                        |
+    | NCLOB                          | string                |                        |
+    | BLOB                           | bytes                 |                        |
+    | BFILE                          | bytes                 |                        |
+
```

Comment:
Have we been using the same table for source and sink? Aren't there some types that are not supported by the sink? How would a BFile work for example? How do other types like interval work? Do they assume the string is a specific format?

Can we document which data types are not supported by the sink? And if there are requirements around what makes a valid string, we need to document those as well, and document what happens if an invalid value is given.

Reply:
I didn't think about this for other DBs, but I imagine we have similar issues in those docs.

Reply:
With the most recent updates Source and Sink support the same data types.
BFile is mapped to the `Types.BYTE` and locator will be written fine as it would be created via CLI using `BFILENAME`  function([from the comment above](https://github.com/data-integrations/database-plugins/pull/46#discussion_r303600292)).

Interval types expect string literals in the standard format, [documented by Oracle](https://docs.oracle.com/database/121/SQLRF/sql_elements003.htm#SQLRF00221).

Added the following comments:
```
| TIMESTAMP WITH TIME ZONE       | string                | Timestamp string in the following format:              |
|                                |                       | "2019-07-15 15:57:46.65 GMT"                           |
| TIMESTAMP WITH LOCAL TIME ZONE | timestamp             |                                                        |
| INTERVAL YEAR TO MONTH         | string                | Oracle's 'INTERVAL YEAR TO MONTH' literal in the       |
|                                |                       | standard format: "year[-month]"                        |
| INTERVAL DAY TO SECOND         | string                | Oracle's 'INTERVAL DAY TO SECOND' literal in the       |
|                                |                       | standard format:                                       |
|                                |                       | "[day] [hour][:minutes][:seconds[.milliseconds]"       |

...
| BFILE                          | bytes                 | BFILE is a data type used to store a locator (link)    |
|                                |                       | to an external file, which is stored outside of the    |
|                                |                       | database. Only the locator will be written to an       |
|                                |                       | Oracle table and not the content of the external file. |
```

Currently, in the case of invalid string literals, pipeline will fail with the following exception:
```
java.lang.IllegalStateException: MapReduce JobId job_local269563738_0003 failed
	at com.google.common.base.Preconditions.checkState(Preconditions.java:176) ~[com.google.guava.guava-13.0.1.jar:na]
	at io.cdap.cdap.internal.app.runtime.batch.MapReduceRuntimeService.run(MapReduceRuntimeService.java:416) ~[na:na]
	at com.google.common.util.concurrent.AbstractExecutionThreadService$1$1.run(AbstractExecutionThreadService.java:52) ~[com.google.guava.guava-13.0.1.jar:na]
	at io.cdap.cdap.internal.app.runtime.batch.MapReduceRuntimeService$2$1.run(MapReduceRuntimeService.java:450) [na:na]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_144]
```
Should this be changed to something meaningful?

> I didn't think about this for other DBs, but I imagine we have similar issues in those docs.

Reference doc of the MS SQL Sink [contains examples](https://github.com/data-integrations/database-plugins/blob/develop/mssql-plugin/docs/SqlServer-batchsink.md#data-types-mapping) of `GEOMETRY` and `GEOGRAPHY` string values, but there is no example of valid `DATETIMEOFFSET` string value.
Also, there are missing mappings in the reference docs for MySQL and Netezza plugins. Should i add them in the scope of this PR?

Reply:
Open separate PRs for the other DB plugins.

Reply:
I used this method: https://docs.oracle.com/database/121/JAJDB/oracle/sql/Datum.html#getBytes__
to
> Create a new Java byte array containing a copy of the RDBMS data.

After that, the bytes were converted back to BFILE instance.

Changed to use https://docs.oracle.com/database/121/JAJDB/oracle/jdbc/OracleBfile.html#getBinaryStream__
to retrieve the entire BFILE as a stream.

Reply:
Ok, I will open separate PRs. Thanks.